tests/test_voting.py

```diff
@@ -110,3 +110,80 @@ def test_consensus_after_non_finalization_streak(casper, funded_privkey, deposit
     casper.vote(mk_suggested_vote(validator_index, funded_privkey))
     assert casper.get_main_hash_justified()
     assert casper.get_votes__is_finalized(casper.get_recommended_source_epoch())
+
+
+def test_logs(casper, funded_privkey, new_epoch, get_logs, deposit_validator,
```

Comment:
I'd rather see this split up into subtests like putting the deposit log test in `test_deposit`.

Not a big deal. I'm planning on doing a bunch of test cleaning/organization if you don't get to it here.

Reply:
We could return to this a bit later 😃 
Checking very field, all Epoch cases etc.
But it will take more time to run